lightningd/watch.c

```diff
@@ -295,3 +305,15 @@ void watch_topology_changed(struct chain_topology *topo)
 		}
 	} while (needs_rerun);
 }
+
+void txwatch_inform(const struct chain_topology *topo,
+		    const struct bitcoin_txid *txid,
+		    const struct bitcoin_tx *tx_may_steal)
+{
+	struct txwatch *txw;
+
+	txw = txwatch_hash_get(&topo->txwatches, txid);
+
+	if (txw && !txw->tx)
+		txw->tx = tal_steal(txw, tx_may_steal);
```

Comment:
In case multiple watchers are watching the same txid (channel factories in the far future, where multiple peer-daemons want to see the confirmation depth of the same tx holding the channel factory), would not this stealing be troublesome? Would not be better to `tal_dup` instead?

Reply:
We need a deep copy, hence the FIXME.  libwally doesn't provide a copy function, which would make it easy to implement bitcoin_tx_dup().

Reply:
A lousy copy would be to serialize, then deserialize, i.e. the "JSON.parse(JSON.stringify())" trick.